package/src/webComponent.ts

```diff
@@ -61,6 +65,14 @@ const addComponent = (name: string, createEditor: typeof basicEditor) => {
 			}
 			set value(value: string) {
 				this.editor.setOptions({ value })
+				this.internals.setFormValue(this.editor.value);
+			}
+
+			formAssociatedCallback() {
+				setTimeout(
+					() => this.internals.setFormValue(this.editor.value),
+					200

```

Comment:
I had to wait ~50ms for the editor's `value` property to be set. I assumed that this delay came from something performance-related, so I opened it up to 200ms.

I think it'd be cool if this could be done with a promise instead, but didn't want to dig that deep without checking with @FIameCaster first.

Reply:
Yeah, this is not an ideal way to do this. When typing in the editor, the internal form value won't be updated. The way to solve both these issues is to add an `update` listener to the editor and set the form value there [like this](https://github.com/FIameCaster/prism-code-editor/pull/20/files#diff-321c11e3e174f1c49325d90f0b9dd8433baf4896e864fcd2b3659332eddbbe7dR51). This will ensure the form value is always in sync. We can also drop the `formAssociatedCallback` then.

I've implemented these changes and it works as expected.

Reply:
Thanks for tidying this up. The perils of jumping into a codebase you're unfamiliar with :sweat_smile: